### PR TITLE
Print messages with zero-length data as if data=None. Addresses #163

### PR DESCRIPTION
**Pull request by radiocane**

### CODE REVIEW

File: can/message.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def __str__(self):
 
         field_strings.append("DLC: {0:d}".format(self.dlc))
         data_strings = []
-        if self.data is not None:
+        if self.data is not None and len(self.data) > 0:
             for index in range(0, self.dlc):
```
Comment (hardbyte) [inline can/message.py]: Isn't the core issue that we are iterating to `dlc` instead of iterating over the data array?

Comment (radiocane) [inline can/message.py]: Yes. I didn't think enough before sending this PR. It'd make more sense to change "range(0, self.dlc)" to "range(0, len(self.data))". Is it ok if I close this PR and open another?

**Comment by christiansandberg** [comment]
Some interfaces use all 8 bytes for the data even if DLC is less than 8 so those should be considered garbage.

**Comment by hardbyte** [comment]
@radiocane you can just commit over top if you like - we can squash it into one commit when merging.

It seems like the number of elements can be given by:
```
min(len(self.data),  self.dlc)
```

The existing `data is not None` check should take care of the remote frame case.

**Comment by radiocane** [comment]
I hope the new version makes more sense :)

**Pull request merged**